src/metabase/api/app.clj

```diff
@@ -68,4 +76,139 @@
   [id]
   (hydrate-details (api/read-check App id)))
 
+(defn- replace-scaffold-targets
+  [structure scaffold-target->target-id]
+  (walk/postwalk
+    (fn [node]
+      (if (and (vector? node) (= "scaffold-target-id" (first node)))
+        (if-let [target-id (get scaffold-target->target-id node)]
+          target-id
+          node)
+        node))
+    structure))
+
+(defn- generate-scaffold
+  [app-name table-ids]
+  (let [tables (hydrate (db/select Table :id [:in table-ids]) :fields)
```

Comment:
- Should we check if all `table-ids` actually point to a table?
- Maybe it would be worth converting `table-ids` into an ordered set, just to make sure we don't generate duplicates. A normal set could be good enough too, we just have to make sure that the first element get's the special treatment it needs.